firewood/src/merkle_util.rs

```diff
@@ -36,12 +36,53 @@ pub enum DataStoreError {
     ProofEmptyKeyValuesError,
 }
 
-pub struct MerkleSetup<S, T> {
+type InMemoryStore = CompactSpace<Node, DynamicMem>;
+
+pub struct InMemoryMerkle<T> {
     root: DiskAddress,
-    merkle: Merkle<S, T>,
+    merkle: Merkle<InMemoryStore, T>,
 }
 
-impl<S: ShaleStore<Node> + Send + Sync, T: BinarySerde> MerkleSetup<S, T> {
+impl<T> InMemoryMerkle<T>
+where
+    T: BinarySerde,
+    EncodedNode<T>: serde::Serialize + serde::de::DeserializeOwned,
```

Comment:
I think this type is over-constrained. There shouldn't be any need to use `BinarySerde` or `EncodedNode<T>` methods in the constructor, should there?

Reply:
Ah you're right... `Merkle::new` is over-constrained... At some point, that should be cleaned up so that you don't need all these extra constraints everywhere, doesn't need to be this PR though